analysers/analyser_osmosis_highway_area_access.py

```diff
@@ -51,24 +51,120 @@
   )
 """
 
+sql20 = """
+CREATE TEMP TABLE {barriertype} AS
+SELECT
+  barrier.id AS id,
+  barrier.tags AS tags,
+  ways.id as wid,
+  ST_AsText(barrier.geom) AS geom
+FROM
+  nodes AS barrier
+  JOIN way_nodes ON
+    way_nodes.node_id = barrier.id
+  JOIN highways AS ways ON
+    ways.id = way_nodes.way_id AND
```

Comment:
It is not intuitive, but this is faster.

```suggestion
  JOIN highways AS ways ON
    barrier.id = ANY (ways.nodes) AND
```

Reply:
Unfortunately this seems to cause a significant slowdown: about a factor 4 for medium-sized countries like netherlands_flevoland, to a factor 300 for a huge country like the (non-split-up) belgium_flanders. Maybe it's only faster if there's only few ways/nodes to compare against? In this sql query it's practically called for every highway way and node as there's no filtering preceding it.